src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -585,6 +589,112 @@ class spell_pri_penance : public SpellScript
     }
 };
 
+// 47758 - Penance (Channel Damage), 47757 - Penance (Channel Healing)
+class spell_pri_penance_channeled : public AuraScript
+{
+    PrepareAuraScript(spell_pri_penance_channeled);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_PRIEST_POWER_OF_THE_DARK_SIDE });
+    }
+
+    void HandleOnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (Unit* caster = GetCaster())
+        {
+            if (caster->HasAura(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE))
+                caster->RemoveAura(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE);
+        }
```

Comment:
```suggestion
        Unit* caster = GetCaster();
        if (!caster || !caster->HasAura(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE))
            return;
            
        caster->RemoveAura(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE);
```

Reply:
Why not just
```cpp
    if (Unit* caster = GetCaster())
        caster->RemoveAura(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE);
```
Or if caster = target
```cpp
    GetTarget()->RemoveAura(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE);
```

Reply:
I thought I was supposed to check if the caster has the aura first before removing it to avoid a crash when attempting to remove it if it was null 🤔

Reply:
crashes during scripting are 99% caused by having `NULL->something()` which is only possible if caster is null in this case. You can simply use RemoveAura, other checks are done inside there

Reply:
> 

yeah GetTarget()->RemoveAura(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE); would be the best solution. I just thought it was a requirement that he has the aura or smth